In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.api as sm

In [6]:
data = pd.read_excel("C:\\Users\\kwame\\Downloads\\users.stat.ufl.edu_~rrandles_sta4210_Rclassnotes_data_textdatasets_KutnerData_Chapter  1 Data Sets_CH01PR19.txt.xlsx", names=["GPA", "ACT"])

In [7]:
# Fit the model
X = data["ACT"]
X = sm.add_constant(X)
y = data["GPA"]
model = sm.OLS(y, X).fit()

##### a. 95% CI on mean GPA for students whose ACT test score is 28 and interpret

In [10]:
X_new = 28

X_new_with_intercept = [1, X_new]  

# Make the prediction
prediction = model.predict(X_new_with_intercept)

print(f"Predicted GPA for ACT = {X_new}: {prediction[0]}")

act_score = 28
confidence_interval = model.get_prediction([1, act_score]).conf_int(alpha=0.05)
print(confidence_interval)

Predicted GPA for ACT = 28: 3.2012088406147248
[[3.06138433 3.34103335]]


We expect students whose ACT test score is 28 to have a cGPA of 3.2, but it could be as low as 3.06 or as high as 3.34.